src/Lecture2.hs

```diff
@@ -85,7 +100,12 @@ lists of even lengths.
 ♫ NOTE: Use eta-reduction and function composition (the dot (.) operator)
   in this function.
 -}
-evenLists = error "TODO"
+
+evenLists :: [[a]] -> [[a]]
+evenLists = filter isLstLenEven
+  where
+    isLstLenEven :: [a] -> Bool  
+    isLstLenEven xs = length xs `mod` 2 == 0 
```

Comment:
You can use standard function `even` to check if the number is even instead of manually dividing by two 🙂 After using `even` you'll be able to write `evenLists` as a one-liner